### PR TITLE
Add headers, navs and user guide boilerplate pages 

### PR DESCRIPTION
**Pull request by seanlowjk**
# Summary 

This PR adds Headers and Navs for easier navigability. It also adds boilerplate code for `Setting Up` and `Introduction` pages in `User Guide`. 

## Description

I have created a separate `ug-nav.md` so that we can leave `site-nav.md` for the main home page, if necessary. 

Also, a link for our `CATcher` repository has been added. And a simple link in `header.md` to direct users to the `Uiser Guide` section. 

I have also created `ug/index.md` and `ug/setting-up.md` as two boilerplate pages on how we should define `frontmatter` for pages in the `User Guide` section. 

## Proposed Commit Message 

```
Add Headers, Navs and User Guide Boilerplate Pages 

There is no indication for the relevant sections in our 
deployed website.

Let's define simple headers and navs to redirect these visitors 
to the necessary sections.  Let's also define some boilerplate 
pages for contributors to follow if they would want to add additional 
sections to the User Guide.  
```

## Screenshots: 

Screenshot from 2021-04-10 15-33-01

Screenshot from 2021-04-10 15-33-10

### CODE REVIEW

File: _markbind/navigation/site-nav.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <navigation>
-* [User Guide]({{ baseUrl }}/index.html)
+* [Home]({{ baseUrl }}/index.html)
```
Comment (ptvrajsk) [inline _markbind/navigation/site-nav.md:2]: What are we planning to put in `Home`? Cuz now like when it starts up its just blank and says home. Should we just have the `ug` as the default?

File: _markbind/headers/header.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 <header>
   <navbar type="dark">
     <a slot="brand" href="{{baseUrl}}/index.html" title="Home" class="navbar-brand"><img src="{{baseUrl}}/assets/catcher_logo.png" width="50"/></a>
+    <a href="{{baseUrl}}/index.html" class="navbar-brand"> Home </a>
+    <a href="{{baseUrl}}/ug/index.html" class="navbar-brand"> User Guide </a>
+    <a href="https://github.com/CATcher-org/CATcher" class="navbar-brand"> CATcher Repo </a>
```
Comment (ptvrajsk) [inline _markbind/headers/header.md:6]: Could we have this look like the icon in here

image

Looks cooler 😅

**Review by ptvrajsk — COMMENTED** [review]
Thanks for adding to this. Can be abit complicated when working with new tools 😅 

Some additional stuff to the review,

Could we try to clean the search result here somehow?

image

So it gives proper sections instead of `index.md` and `settingup.md` or is that not possible? I'm not entirely sure yet still figuring this out too.

**Comment by seanlowjk** [comment]
> Could we try to clean the search result here somehow?

Eyes, we should be able to do so using `site.json`. Let me clean it up and post an update here when I'm done!

File: _markbind/headers/header.md (hunk 2 above)
Comment (seanlowjk) [reply _markbind/headers/header.md:6]: Screenshot from 2021-04-10 22-49-35

Will this be Ok? :sweat_smile:

File: _markbind/navigation/site-nav.md (hunk 1 above)
Comment (seanlowjk) [reply _markbind/navigation/site-nav.md:2]: I think we can fill in like a brief description of our product like what [Reposense](https://reposense.org/) has done XD

File: ug/setting-up.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  footer: footer.md
+</frontmatter>
+
+# Get started
+
+You can either use CATcher through your web browser, or download CATcher's desktop application.
+
+Once the app is launched (either web or desktop version), it will prompt you to enter the session you are participating in, using a dropdown.
+
+<img width="100%" src="https://imgur.com/nBOy7zH.png" />
```
[the remark this answers is not in the document]
Comment (seanlowjk) [reply ug/setting-up.md:15]: Weird... I think I will just download the image and upload it to the `assets` folder

File: ug/index.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  header: header.md
+  pageNav: 2
+  pageNavTitle: "Chapters of This Page"
+  siteNav: ug-nav.md
+  footer: footer.md
+</frontmatter>
+
+# Welcome to the CATcher user guide
+
+CATcher is a desktop application for Crowd-sourced Anonymous Testing software. It uses GitHub as the backend for hosting bug reports.
```
[the remark this answers is not in the document]
Comment (seanlowjk) [reply ug/index.md:11]: Yes definitely haha, we will add more in future PRs :smile: !

File: ug/setting-up.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  header: header.md
+  pageNav: 2
+  pageNavTitle: "Chapters of This Page"
+  siteNav: ug-nav.md
+  footer: footer.md
+</frontmatter>
+
+# Get started
+
+You can either use CATcher through your web browser, or download CATcher's desktop application.
```
[the remark this answers is not in the document]
Comment (seanlowjk) [reply ug/setting-up.md:11]: Sure! That's a really good idea :100:

File: ug/index.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<frontmatter>
+  header: header.md
+  pageNav: 2
+  pageNavTitle: "Chapters of This Page"
+  siteNav: ug-nav.md
+  footer: footer.md
+</frontmatter>
+
+# Welcome to the CATcher user guide
```
[the remark this answers is not in the document]
Comment (seanlowjk) [reply ug/index.md:9]: Screenshot from 2021-04-10 23-10-08

:+1:

File: _markbind/navigation/site-nav.md (hunk 1 above)
Comment (ptvrajsk) [reply _markbind/navigation/site-nav.md:2]: Ahh okay cool!

**Review by ptvrajsk — APPROVED** [review]
Thanks for working on this man, I think its some good boilerplate content that we can refine as we go along 👍

File: ug/index.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+CATcher is a desktop application for Crowd-sourced Anonymous Testing software. It uses GitHub as the backend for hosting bug reports.
```
Comment (anubh-v) [inline ug/index.md:11]: Minor grammar fix:

```suggestion
CATcher is a desktop application for Crowd-sourced Anonymous Testing of software. It uses GitHub as the backend for hosting bug reports.
```

**Review by anubh-v — APPROVED** [review]
Thanks @seanlowjk , the base structure LGTM

File: ug/index.md (hunk 7 above)
Comment (seanlowjk) [reply ug/index.md:11]: Alright!

Comment (anubh-v) [reply ug/index.md]: Sorry, can you also change `... is a desktop application` to `...is a cross platform application` ?

Comment (seanlowjk) [reply ug/index.md]: Oops :x missed that out .. alright

**Pull request merged**